docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/builder/DockerBuilderPublisher.java

```diff
@@ -25,34 +21,88 @@
 import org.apache.commons.io.IOUtils;
 import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.plugins.tokenmacro.MacroEvaluationException;
 import org.jenkinsci.plugins.tokenmacro.TokenMacro;
 import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+import shaded.com.google.common.base.Joiner;
+import shaded.com.google.common.base.Optional;
+import shaded.com.google.common.base.Splitter;
+import shaded.com.google.common.base.Throwables;
 
+import javax.annotation.CheckForNull;
 import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
 
 /**
  * Builder extension to build / publish an image from a Dockerfile.
  */
 public class DockerBuilderPublisher extends Builder implements Serializable {
+    private static final Pattern VALID_REPO_PATTERN = Pattern.compile("^([a-z0-9-_.]+)$");
 
     public final String dockerFileDirectory;
-    public final String tag;
+
+    /**
+     * @deprecated use {@link #tags}
+     */
+    @Deprecated
+    public String tag;
+
+    @CheckForNull
+    private List<String> tags;
+
     public final boolean pushOnSuccess;
     public final boolean cleanImages;
     public final boolean cleanupWithJenkinsJobDelete;
 
     @DataBoundConstructor
-    public DockerBuilderPublisher(String dockerFileDirectory, String tag, boolean pushOnSuccess, boolean cleanImages, boolean cleanupWithJenkinsJobDelete) {
+    public DockerBuilderPublisher(String dockerFileDirectory,
+                                  String tagsString,
+                                  boolean pushOnSuccess,
+                                  boolean cleanImages,
+                                  boolean cleanupWithJenkinsJobDelete) {
         this.dockerFileDirectory = dockerFileDirectory;
-        this.tag = tag;
+        setTagsString(tagsString);
+        this.tag = null;
         this.pushOnSuccess = pushOnSuccess;
         this.cleanImages = cleanImages;
         this.cleanupWithJenkinsJobDelete = cleanupWithJenkinsJobDelete;
     }
 
+    public List<String> getTags() {
+        return tags;
+    }
+
+    public void setTags(List<String> tags) {
+        this.tags = tags;
+    }
+
+    public String getTagsString() {
+        return getTags() == null ? "" : Joiner.on("\n").join(getTags());
+    }
+
+    public void setTagsString(String tagsString) {
+        setTags(filterStringToList(tagsString));
+    }
+
+    public static List<String> filterStringToList(String str) {
+        return str == null ? null : Splitter.on("\n").omitEmptyStrings().trimResults().splitToList(str);
+    }
+
+    public static void verifyTags(String tagsString) {
+        final List<String> verifyTags = filterStringToList(tagsString);
+        for (String verifyTag : verifyTags) {
+            if (!VALID_REPO_PATTERN.matcher(verifyTag).matches()) {
+                throw new IllegalArgumentException("Tag " + verifyTag + " doesn't match ^([a-z0-9-_.]+)$");
```

Comment:
can you use const here? Or instead you can change const in future, but can forget to change it here.